stdlib/InteractiveUtils/src/editless.jl

```diff
@@ -82,15 +84,16 @@ already work:
 The following defines the usage of terminal-based `emacs`:
 
     define_editor(
-        r"\\bemacs\\b.*\\s(-nw|--no-window-system)\\b", wait=true) do cmd, path, line
-        `\$cmd +\$line \$path`
+        r"\\bemacs\\b.*\\s(-nw|--no-window-system)\\b", wait=true
+    ) do cmd, path, line, wait
+        run(pipeline(`\$cmd +\$line \$path`); wait)
```

Comment:
But that's not right it doesn't match the actual `define_editor` calls in this file (which you did not change).

Besides, `define_editor` of course should *not* immediately run the editor, it is supposed to return a `Cmd` object.

And just passing `wait=true` to `run` won't in fact cause the editor to "wait" in general: for those commands which open a GUI editor window outside the terminal, sometimes there exists a command line argument to allow choosing between "open an editor window and immediately exit this launcher" and "open an editor window and wait until the users closes it".

E.g. for `code` (VS Code) and `bbedit` it always exits immediately, unless one uses the `--wait` argument.

Reply:
You are right! I misread the docs for this. I've reverted the example back to one without any `wait`. 

This PR doesn't change the behavior for any editors. It'll just allow supporting `code --wait` for specific cases.